src/PrestaShopBundle/Form/Admin/Improve/International/Locations/StateType.php

```diff
@@ -0,0 +1,170 @@
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Form\Admin\Improve\International\Locations;
+
+use PrestaShop\PrestaShop\Core\ConstraintValidator\Constraints\CleanHtml;
+use PrestaShop\PrestaShop\Core\ConstraintValidator\Constraints\TypedRegex;
+use PrestaShop\PrestaShop\Core\ConstraintValidator\Constraints\UniqueStateIsoCode;
+use PrestaShop\PrestaShop\Core\Domain\State\StateSettings;
+use PrestaShop\PrestaShop\Core\Domain\State\ValueObject\StateId;
+use PrestaShop\PrestaShop\Core\Form\ConfigurableFormChoiceProviderInterface;
+use PrestaShopBundle\Form\Admin\Type\ConfigurableCountryChoiceType;
+use PrestaShopBundle\Form\Admin\Type\SwitchType;
+use PrestaShopBundle\Form\Admin\Type\ZoneChoiceType;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Translation\TranslatorInterface;
+use Symfony\Component\Validator\Constraints\Length;
+use Symfony\Component\Validator\Constraints\NotBlank;
+
+class StateType extends AbstractType
+{
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+    /**
+     * @var ConfigurableFormChoiceProviderInterface
+     */
+    private $countriesChoiceProvider;
+    /**
+     * @var ConfigurableFormChoiceProviderInterface
+     */
+    private $zonesChoiceProvider;
+
+    /**
+     * StateType constructor.
+     *
+     * @param TranslatorInterface $translator
+     * @param ConfigurableFormChoiceProviderInterface $countriesChoiceProvider
+     * @param ConfigurableFormChoiceProviderInterface $zonesChoiceProvider
+     */
+    public function __construct(
+        TranslatorInterface $translator,
+        ConfigurableFormChoiceProviderInterface $countriesChoiceProvider,
+        ConfigurableFormChoiceProviderInterface $zonesChoiceProvider
+    ) {
+        $this->translator = $translator;
+        $this->countriesChoiceProvider = $countriesChoiceProvider;
+        $this->zonesChoiceProvider = $zonesChoiceProvider;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $stateIdValue = isset($builder->getData()['id_state']) && $builder->getData()['id_state'] instanceof StateId ?
+            $builder->getData()['id_state']->getValue() :
+            null;
+
+        $builder
+            ->add('name', TextType::class, [
+                'required' => true,
+                'help' => $this->translator->trans('Provide the state name to be displayed in addresses and on invoices', [], 'Admin.International.Help'),
```

Comment:
```suggestion
                'help' => $this->translator->trans('Provide the state name to be displayed in addresses and on invoices.', [], 'Admin.International.Help'),
```

The missing period is creating a new string.

Reply:
Fixed